### ISSUE TITLE
Object animation/rotation.

### ISSUE THREAD

**Issue by Beliaar** [unknown-horizons/godot-port#24]
The current gfx of UH were made for a purely 2D engine. The animations are splitted into 4 (buildings) and 8 (units) directions respectively. The engine would take care of using the correct direction path for the given rotation of an object.

We need to find a way to either use the current prerendered images with godot  or, since we have the original blender files available, reexport the graphics in a way that they are better usable with godot.

For the first we need to either find or make a class/script that can calculate the image to be used for any given angle.

For the second we would need to evaluate what we can actually do with the current blender files.

**Comment by aaronfranke** [comment]
I'm personally a fan of 3D, but that's up to @LinuxDonald

**Comment by LinuxDonald** [comment]
The game is made in 2.5D i think we should do it again in 2.5D because of the look and feel.

**Comment by Beliaar** [comment]
Well, the game logic can still be 2.5D. But it does not matter for me.

**Comment by LinuxDonald** [comment]
Would it make an visible difference when we would do the graphics in 3D?

**Comment by Beliaar** [comment]
Well, if we use the same assets it should not. It makes it easier to have things rotated in our case, for example, though.

There is are advantages to both 2D and 3D has, but it's not really something that I know in enough detail to explain.

**Comment by LinuxDonald** [comment]
3D would be easier because of graphics. On 2D we would need a lot of files or Atlas for each animation/building and so on.

**Comment by LinuxDonald** [comment]
@SubSage what do you think about this?

**Comment by jmdejong** [comment]
There's also https://github.com/unknown-horizons/unknown-horizons/issues/2814

Using 3d models could solve this, though very fancy 2d could do that too.

Do we have 3d models of all assets? I thought there were a few that we only had in 2d.

**Comment by nihathrael** [comment]
I think all the ground tiles are handmade sprites iirc.

**Comment by aaronfranke** [comment]
Yes, using 3D models would completely solve that problem. We can use 3D even if some things are just "billboards" i.e. 2D objects always facing the camera. Buildings and ships should probably be full 3D objects but it should be fine if characters or terrain have no depth.

**Comment by LinuxDonald** [comment]
@nihathrael have you maybe discord? I could need for help about code questions and UH.

**Comment by Beliaar** [comment]
Yeah, tiles would not be the problem.

**Comment by chiniara** [comment]
Making a mixed approach would benefit the game and fix some problems you have.
The world can be 3D but everything else could use billboards so you don't need to re-render all your stuff

Rotating, z sorting issues.

If you want to make the buildings 3D probably some retopology would be needed because from what i see the models are fairly highpoly

**Comment by nitramr** [comment]
Hey, I made a quick test for usage of 3D models. 

![bildschirmfoto von 2019-02-03 18-27-29](https://user-images.githubusercontent.com/15112256/52180038-77a5f300-27e1-11e9-8c2e-440ac5220629.png)

Models are quiet good in resolution, but the material needs much more effort to prepare as you can see in the image. 
Godot works well with Collada files. It is recommended to use a special exporter in Blender. See that link below.
https://godotengine.org/article/importing-3d-assets-blender-gamedevtv

To get shadows in the model I set an extra sun light object to the world scene.

**Comment by Jummit** [comment]
I think a pixelation shader plus 3d models would make it feel like the original, but make rotation easier.

**Comment by brylie** [comment]
I am available to help with 3D models, where they are lacking. Is there a tracker task for porting all 2.5D models to 3D?

I would really like to see this game in 3D, with free rotating camera controlled by mouse :-)

**Comment by aaronfranke** [comment]
@brylie We can make a tracker if we have progress to put on it.

**Comment by brylie** [comment]
Sure. A list of items to be ported to 3D would be a good start, or at least a pointer to an assets folder containing the items. Then we can divide the work.
